rasa/nlu/utils/spacy_utils.py

```diff
@@ -88,21 +83,82 @@ def cache_key(
         cls, component_meta: Dict[Text, Any], model_metadata: "Metadata"
     ) -> Optional[Text]:
 
-        # Fallback, use the language name, e.g. "en",
-        # as the model name if no explicit name is defined
-        spacy_model_name = component_meta.get("model", model_metadata.language)
+        spacy_model_name = cls.check_model_fallback(
+            component_meta.get("model"), model_metadata.language, warn=True
+        )
 
         return cls.name + "-" + spacy_model_name
 
+    @classmethod
+    def check_model_fallback(cls, spacy_model_name, language_name, warn=False):
```

Comment:
let's make this protected as it shouldn't be used outside of it.
```suggestion
    def _check_model_fallback(cls, spacy_model_name, language_name, warn=False):
```